nixos/modules/services/monitoring/incron.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.incron;
+
+in
+
+{
+  options = {
+
+    services.incron = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the incron daemon.";
+      };
+
+      allow = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
```

Comment:
Is there some reason for using null instead of an empty list?

Reply:
Yes. In the case that allow is null no file will be created. In the case that an empty array is passed an empty file would be created. These two situations have a different outcome.

An empty allow file means all users are allowed to have their own incrontab.

Reply:
Do you mean an empty deny? But you are right according to the [man page](https://github.com/ar-/incron/blob/1eedfbc9b318372efd119fd17f4abdbde561a53d/incrond.8#L18). You should probably mention it in the option description.

Reply:
Good point. I had modified the description. I hope the intent is clear enough.